### PR TITLE
Enable Free Mints

### PR DESCRIPTION
**Pull request by freddyaboulton**
Add a separate whitelist for free mints.

### CODE REVIEW

File: pmv-eth/contracts/PMVMixin.sol [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// contracts/PMV.sol
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
```
Comment (freddyaboulton) [inline pmv-eth/contracts/PMVMixin.sol:4]: To reduce the code dup between the two contracts

File: pmv-eth/contracts/PMV.sol [hunk 2]
```diff
@@ -43,6 +36,19 @@ contract PMV is ERC721Enumerable, Ownable {
         presaleMints[msg.sender] += tokenQuantity;      
     }
 
+    function mintFree(uint256 allowance, bytes32[] calldata proof, uint256 tokenQuantity) external {
+        require(freeMintAllowed, "Free mint not allowed");
+        require(proof.verify(rootMintFree, keccak256(abi.encodePacked(msg.sender, allowance))), "NOT ON WHITELIST");
```
Comment (chrishol) [inline pmv-eth/contracts/PMV.sol:41]: if you care, would consider using allowlist over whitelist wherever we use it

File: pmv-eth/contracts/PMV.sol [hunk 3]
```diff
@@ -43,6 +36,19 @@ contract PMV is ERC721Enumerable, Ownable {
...
+        require(freeMints[msg.sender] + tokenQuantity <= allowance, "MINTING MORE THAN ALLOWED");
+        require(tokenQuantity + totalSupply() <= maxSupply, "NOT ENOUGH LEFT IN STOCK");
+        
+        for(uint256 i = 0; i < tokenQuantity; i++) {
+            _mint(msg.sender, totalSupply() + 1);
```
Comment (chrishol) [inline pmv-eth/contracts/PMV.sol:46]: I don't think we should call `totalSupply()` each time, as it reads for storage every time - we should read it once at the start of the method and then use it in all the different places.

Here's an example (L93): https://etherscan.io/address/0xe844574d8ddcb6170840f8ebd57d6c94e167656c#code

File: pmv-eth/contracts/PMV.sol (hunk 2 above)
Comment (freddyaboulton) [reply pmv-eth/contracts/PMV.sol:41]: Done!  https://github.com/freddyaboulton/pmv-token/pull/61

File: pmv-eth/contracts/PMV.sol (hunk 3 above)
Comment (freddyaboulton) [reply pmv-eth/contracts/PMV.sol:46]: Done! https://github.com/freddyaboulton/pmv-token/pull/61

**Pull request merged**